lib/hawkbit/hawkbit.c

```diff
@@ -0,0 +1,1212 @@
+	};
+
+	if (setsockopt(hb_context.sock, SOL_TLS, TLS_SEC_TAG_LIST, sec_tag_opt,
+		       sizeof(sec_tag_opt)) < 0) {
+		LOG_ERR("Failed to set TLS_TAG option");
+		return false;
+	}
+
+	if (setsockopt(hb_context.sock, SOL_TLS, TLS_HOSTNAME, HAWKBIT_SERVER,
+		       sizeof(HAWKBIT_SERVER)) < 0) {
+		return false;
+	}
+#endif
+
+	if (connect(hb_context.sock, addr->ai_addr, addr->ai_addrlen) < 0) {
+		LOG_ERR("Failed to connect to Server");
+		return false;
+	}
+
+	return true;
+}
+
+static void cleanup_connection(void)
+{
+	if (close(hb_context.sock) < 0) {
+		LOG_ERR("Could not close the socket");
+	}
+}
+
+static int hawkbit_time2sec(const char *s)
+{
+	int sec = 0;
+
+	/* Time: HH:MM:SS */
+	sec = strtol(s, NULL, 10) * (60 * 60);
+	sec += strtol(s + 3, NULL, 10) * 60;
+	sec += strtol(s + 6, NULL, 10);
+
+	if (sec < 0) {
+		return -1;
+	} else {
+		return sec;
+	}
+}
+
+static const char *hawkbit_status_finished(enum hawkbit_status_fini f)
+{
+	switch (f) {
+	case HAWKBIT_STATUS_FINISHED_SUCCESS:
+		return "success";
+	case HAWKBIT_STATUS_FINISHED_FAILURE:
+		return "failure";
+	case HAWKBIT_STATUS_FINISHED_NONE:
+		return "none";
+	default:
+		LOG_ERR("%d is invalid", (int)f);
+		return NULL;
+	}
+}
+
+static const char *hawkbit_status_execution(enum hawkbit_status_exec e)
+{
+	switch (e) {
+	case HAWKBIT_STATUS_EXEC_CLOSED:
+		return "closed";
+	case HAWKBIT_STATUS_EXEC_PROCEEDING:
+		return "proceeding";
+	case HAWKBIT_STATUS_EXEC_CANCELED:
+		return "canceled";
+	case HAWKBIT_STATUS_EXEC_SCHEDULED:
+		return "scheduled";
+	case HAWKBIT_STATUS_EXEC_REJECTED:
+		return "rejected";
+	case HAWKBIT_STATUS_EXEC_RESUMED:
+		return "resumed";
+	case HAWKBIT_STATUS_EXEC_NONE:
+		return "none";
+	default:
+		LOG_ERR("%d is invalid", (int)e);
+		return NULL;
+	}
+}
+
+/**
+ * @brief Update an ACID of a given type on flash
+ *
+ * @param type ACID type to update
+ * @param acid New ACID value
+ * @return 0 on success, negative on error.
+ */
+static int hawkbit_device_acid_update(enum hawkbit_dev_acid_t type,
+					u32_t new_value)
+{
+	struct hawkbit_device_acid device_action_id;
+	int ret;
+
+	ret = nvs_read(&fs, ADDRESS_ID, &device_action_id,
+		       sizeof(device_action_id));
+	if (ret < 0) {
+		LOG_ERR("Failed to read device action id");
+		return ret;
+	}
+
+	if (type == HAWKBIT_ACTION_ID_UPDATE) {
+		device_action_id.update = new_value;
+	} else {
+		device_action_id.current = new_value;
+	}
+
+	ret = nvs_write(&fs, ADDRESS_ID, &device_action_id,
+			sizeof(device_action_id));
+	if (ret < 0) {
+		LOG_ERR("Failed to write device id");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/*
+ * Update sleep interval, based on results from hawkbit base polling
+ * resource
+ */
+static void hawkbit_update_sleep(struct hawkbit_ctl_res *hawkbit_res)
+{
+	const char *sleep = hawkbit_res->config.polling.sleep;
+	int len;
+
+	if (strlen(sleep) != HAWKBIT_SLEEP_LENGTH) {
+		LOG_ERR("Invalid poll sleep: %s", sleep);
+	} else {
+		len = hawkbit_time2sec(sleep);
```

Comment:
Instead of `len` do `sleep_time` or something like that. The len is typically used for string lengths